wandb/cli/cli.py

```diff
@@ -966,72 +949,138 @@ def launch_sweep(
+                )
+                return False
+
+        # grab the queue from previously run scheduler if not specified
+        if not queue and prev_sweep_run_spec.get("overrides", {}).get(
+            "run_config", {}
+        ).get("sweep_args", {}).get("queue"):
+            queue = prev_sweep_run_spec["overrides"]["run_config"]["sweep_args"][
+                "queue"
+            ]
+
+    if not queue:
+        wandb.termerror(
+            "Launch-sweeps require setting a 'queue', use --queue option or a 'queue' key in the 'launch' section in the config"
+        )
+        return
 
+    entrypoint = Scheduler.ENTRYPOINT if not scheduler_job else None
     args = sweep_utils.construct_scheduler_args(
-        sweep_config=parsed_sweep_config,
+        is_job=scheduler_job is not None,
+        sweep_config=sweep_config,
         queue=queue,
         project=project,
         author=entity,
     )
     if not args:
-        # error already logged
         return
 
-    overrides = {
-        "overrides": {
-            "run_config": {
-                "scheduler": scheduler_args,
-                "launch": launch_args,
-            },
-            "args": args,
-        }
-    }
+    # validate training job existence
+    if not sweep_utils.check_job_exists(PublicApi(), sweep_config.get("job")):
+        return False
+
+    # validate scheduler job existence, if present
+    if not sweep_utils.check_job_exists(PublicApi(), scheduler_job):
+        return False
+
+    # Set run overrides for the Scheduler
+    overrides = {"run_config": {}}
+    if launch_args:
+        overrides["run_config"]["launch"] = launch_args
+    if scheduler_args:
+        overrides["run_config"]["scheduler"] = scheduler_args
+    if settings:
+        overrides["run_config"]["settings"] = settings
+
+    if scheduler_job:
+        overrides["run_config"]["sweep_args"] = args
+    else:
+        overrides["args"] = args
+
+    # set name of scheduler
+    prefix = "sweep" if scheduler_job else "wandb"
+    name = f"{prefix}-scheduler-WANDB_SWEEP_ID"
+    if scheduler_args.get("name"):
+        name = scheduler_args["name"]
 
     # Launch job spec for the Scheduler
     launch_scheduler_spec = construct_launch_spec(
         uri=Scheduler.PLACEHOLDER_URI,
         api=api,
-        name="Scheduler.WANDB_SWEEP_ID",
+        name=name,
         project=project,
         entity=entity,
         docker_image=scheduler_args.get("docker_image"),
         resource=scheduler_args.get("resource", "local-process"),
-        entry_point=Scheduler.ENTRYPOINT,
+        entry_point=entrypoint,
         resource_args=scheduler_args.get("resource_args", {}),
         repository=launch_args.get("registry", {}).get("url", None),
-        job=None,
+        job=scheduler_job,
         version=None,
-        launch_config=overrides,
-        run_id=None,
+        launch_config={"overrides": overrides},
+        run_id="WANDB_SWEEP_ID",  # scheduler inits run with sweep_id=run_id
```

Comment:
Wait, I thought this was not necessary anymore?

Reply:
The scheduler can fetch the sweep id from its call to upsert sweep no?

Reply:
The problem is that if we don't set the run_id, we dump a new RUN_ID into the environment vars, which overwrites the call to wandb.init in the scheduler... So even `wandb.init(id=self._sweep_id)` actually ignores that passed in `id` and uses `WANDB_RUN_ID` passed in through launch. Suggestions? @KyleGoyette

Reply:
I see 👍